modules/configuration/src/test/java/org/apache/ignite/internal/configuration/util/ConfigurationUtilTest.java

```diff
@@ -204,29 +204,94 @@ public void findSuccessfully() {
                 )
         );
 
+        NodeValue<Object> root = find(List.of(), parentNode, true);
+        assertNull(root.field());
+        assertSame(parentNode, root.value());
+
+        NodeValue<Object> elements = find(List.of("elements"), parentNode, true);
+        assertNotNull(elements.field());
+        assertSame(parentChange.elements(), elements.value());
+
+        NodeValue<Object> elementsName = find(List.of("elements", "name"), parentNode, true);
+        assertNotNull(elementsName.field());
+        assertSame(parentChange.elements().get("name"), elementsName.value());
+
+        NodeValue<Object> elementsNameChild = find(List.of("elements", "name", "child"), parentNode, true);
+        assertNotNull(elementsNameChild.field());
+        assertSame(parentChange.elements().get("name").child(), elementsNameChild.value());
+
+        NodeValue<Object> elementsNameChildStr = find(List.of("elements", "name", "child", "str"), parentNode, true);
+        assertNotNull(elementsNameChildStr.field());
+        assertSame(parentChange.elements().get("name").child().str(), elementsNameChildStr.value());
+    }
+
+    @Test
+    public void findSuccessfullyPolymorphicConfig() {
```

Comment:
We need a test where polymorphic extensions would have a sub-configuration with the same name, that is not primitive in both cases. I bet it wouldn't work.